policies/verify-tenant-patch/policies/rego_rule_test.rego

```diff
@@ -0,0 +1,398 @@
+package crownlabstenantconstraint
+
+test_invalid_patch_basic_user {
+	input := {"review": input_review1}
+	result := violation with input as input
+	count(result) > 0
+}
+
+test_invalid_patch_email {
+	input := {"review": input_review5}
+	result := violation with input as input
+	count(result) > 0
+}
+
+test_valid_patch_basic_user {
+	input := {"review": input_review2}
+	result := violation with input as input
+	count(result) == 0
+}
+
+test_object_creation {
+	input := {"review": input_review3}
+	result := violation with input as input
+	count(result) == 0
+}
+
+test_valid_patch_admin_user {
+	input := {"review": input_review4}
+	result := violation with input as input
+	count(result) == 0
+}
+
+# invalid patch for user: can't change the role
+input_review1 = output {
+	output = {
+		"resource": {"group": "crownlabs.polito.it", "version": "v1alpha1", "resource": "tenants"},
+		"requestKind": {"kind": "Tenant", "group": "crownlabs.polito.it", "version": "v1alpha1"},
+		"operation": "UPDATE",
+		"userInfo": {"username": "minikube-user", "groups": ["system:masters", "system:authenticated"]},
+		"oldObject": {
+			"apiVersion": "crownlabs.polito.it/v1alpha1",
+			"kind": "Tenant",
+			"metadata": {
+				"uid": "664a606d-0f45-4ca3-9ae5-629bc86495eb",
+				"creationTimestamp": "2020-12-20T11:14:59Z",
+				"generation": 1,
+				"name": "example.com",
+				"resourceVersion": "231625",
+			},
+			"spec": {
+				"name": "test-name",
+				"publicKeys": ["test-key1 test-key2"],
+				"surname": "test-surname",
+				"workspaces": [{"groupNumber": 1, "role": "Basic", "workspaceURL": "www.test.it"}],
+				"ID": "s00000",
+				"createSandbox": false,
+				"email": "testemail@studenti.polito.it",
+			},
+		},
+		"options": {"kind": "UpdateOptions", "apiVersion": "meta.k8s.io/v1", "fieldManager": "kubectl-patch"},
+		"_unstable": {},
+		"uid": "3771271d-9194-4924-b3b6-edccf84d8f76",
+		"requestResource": {"group": "crownlabs.polito.it", "version": "v1alpha1", "resource": "tenants"},
+		"name": "example.com",
+		"object": {
+			"apiVersion": "crownlabs.polito.it/v1alpha1",
+			"kind": "Tenant",
+			"metadata": {
+				"uid": "664a606d-0f45-4ca3-9ae5-629bc86495eb",
+				"creationTimestamp": "2020-12-20T11:14:59Z",
+				"generation": 1,
+				"managedFields": [
+					{
+						"operation": "Update",
+						"time": "2020-12-20T11:14:59Z",
+						"apiVersion": "crownlabs.polito.it/v1alpha1",
+						"fieldsType": "FieldsV1",
+						"fieldsV1": {"f:spec": {".": {}, "f:ID": {}, "f:createSandbox": {}, "f:name": {}, "f:publicKeys": {}, "f:surname": {}, "f:workspaces": {}}}, "manager": "kubectl-create",
+					},
+					{
+						"fieldsType": "FieldsV1", "fieldsV1": {"f:spec": {"f:email": {}}},
+						"manager": "kubectl-patch",
+						"operation": "Update",
+						"time": "2020-12-20T11:30:12Z",
+						"apiVersion": "crownlabs.polito.it/v1alpha1",
+					},
+				],
```

Comment:
You can drop the managedFields, to shorten this file. The same for all occurrences